cmd/modelcmd/filesystem.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelcmd
+
+import (
+	"io"
+	"os"
+
+	"github.com/juju/errors"
+)
+
+// ReadSeekCloser is a minimal interface used for os.File.
+type ReadSeekCloser interface {
+	io.ReadCloser
+	io.Seeker
+}
+
+// Filesystem is an interface providing access to the filesystem,
+// either delegating to calling os functions or functions which
+// always return an error.
+type Filesystem interface {
```

Comment:
probably consider to gomock this for cmd tests in the future

Reply:
We can mock later when it may be needed. The CLI tests pretty much all set up a fake Juju home and access files from there. So at the moment,, there's no need to mock anything as everything is done in a tmp dir.